projects/cardiovascular-patient-app/src/app/report-bp/custom-error-state-matcher.spec.ts

```diff
@@ -0,0 +1,39 @@
+import { FormControl, FormGroupDirective, NgForm, Validators } from '@angular/forms';
+import { CustomErrorStateMatcher } from './custom-error-state-matcher';
+
+describe('CustomErrorStateMatcher', () => {
+    let customErrorStateMatcher: CustomErrorStateMatcher;
+
+    beforeEach(() => {
+        customErrorStateMatcher = new CustomErrorStateMatcher();
+    });
+
+    it('should return false if form is submitted', () => {
+        const control = new FormControl();
+        const form = {
+            submitted: true
+        } as FormGroupDirective | NgForm;
+
+        expect(customErrorStateMatcher.isErrorState(control, form)).toBe(false);
+    });
```

Comment:
This is not exactly the behavior that we want. I think we need multiple test cases:
```
it('should return false if form is untouched after being submitted')
it('should return true if form is touched and invalid after being submitted')
```

Reply:
Okay @semicolin